### PR TITLE
CSP voting

### PR DESCRIPTION
**Pull request by nigeon**
Signer generation on Client Provider
Added CSP (oAuth) voting to the Election component

### CODE REVIEW

**Comment by nigeon** [comment]
Hi! Is there any missing stuff for this to be merged?

File: packages/chakra-components/src/components/Election/Election.tsx [hunk 1]
```diff
@@ -91,17 +102,19 @@ export const useElectionProvider = ({
       setVoterAddress(address)
       const isIn = await client.isInCensus(election.id)
       setIsInCensus(isIn)
+      const censusType: CensusType = election.census.type
+      setCensusType(censusType)
 
       let left = 0
-      if (isIn) {
+      if (isIn || censusType == CensusType.WEIGHTED) {
```
Comment (elboletaire) [inline packages/chakra-components/src/components/Election/Election.tsx:109]: use strict check please `===`

File: packages/chakra-components/src/components/Election/Election.tsx [hunk 2]
```diff
@@ -91,17 +102,19 @@ export const useElectionProvider = ({
...
         // no need to check votes left if member ain't in census
         left = await client.votesLeftCount(election.id)
         setVotesLeft(left)
 
         const voted = await client.hasAlreadyVoted(election.id)
         setVoted(voted)
       }
-      setIsAbleToVote(left > 0 && isIn)
+      setIsAbleToVote((left > 0 && isIn) || censusType == CensusType.CSP)
```
Comment (elboletaire) [inline packages/chakra-components/src/components/Election/Election.tsx:117]: Same here, unless it's intended (I doubt it for this case), use strict check always.

**Review by elboletaire — CHANGES_REQUESTED** [review]

**Review by elboletaire — APPROVED** [review]
Code LGTM. Let me know if I should try it by myself as we did speak, or if you consider it good to go (I'd merge it and create a release in that case).

**Pull request merged**